tests/test_middleware.py

```diff
@@ -297,3 +297,28 @@ async def handler(request):
 
     _, response = app.test_client.get("/")
     assert response.json["foo"] == "bar"
+
+
+def test_middleware_added_response(app):
+    response_middleware_run_count = 0
+    request_middleware_run_count = 0
+
+    @app.on_response
+    def response(_, response):
+        nonlocal response_middleware_run_count
+        response_middleware_run_count += 1
+
+    @app.on_request
+    def request(_):
+        nonlocal request_middleware_run_count
+        request_middleware_run_count += 1
+
+    @app.get("/")
+    async def handler(request):
+        resp1 = await request.respond()
+        await resp1.eof()
+        return text("")
```

Comment:
Food for thought, maybe we should never allow a return on a `respond` handler.

Reply:
Got it, will rewrite the tests

Reply:
Considered as resolved.